[Power Testing] WebKit.GPU cpu increase in fullscreen video playback on intel[2% power, 12 mins]

jyavenard opened the pull request:
#### 8b858a5c1c346a4c79899586b752ca7845049f70
<pre>
[Power Testing] WebKit.GPU cpu increase in fullscreen video playback on intel[2% power, 12 mins]
<a href="https://bugs.webkit.org/show_bug.cgi?id=297102">https://bugs.webkit.org/show_bug.cgi?id=297102</a>
<a href="https://rdar.apple.com/157144345">rdar://157144345</a>

Reviewed by Abrar Rahman Protyasha.

Disable the use a VTDecompressionSession for MSE on Intel platform as it
causes increased power usage.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:

Canonical link: <a href="https://commits.webkit.org/298484@main">https://commits.webkit.org/298484@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dd1c8af2fd08df80efebb3677d1cd6ca8d767c60

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115408 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35102 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25604 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121497 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66122 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/823f210e-e255-4ff9-acc4-7d2a9edeaba5) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35761 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43673 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87688 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/42453 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/70f05ded-eb5d-46d4-b675-89c2e2f477cd) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118356 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28522 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103593 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68080 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27680 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21715 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65151 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107615 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97911 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21828 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124662 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113947 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42347 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31722 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96466 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42715 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99781 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96252 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24530 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41484 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19339 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38253 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42227 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47783 "Built successfully") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/139047 "Build was cancelled. Recent messages:") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41729 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/139047 "Build was cancelled. Recent messages:") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45057 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43448 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5043,8 +5043,10 @@ MediaSourcePrefersDecompressionSession:
+      "CPU(X86_64) || CPU(X86)": false
       default: true
     WebCore:
+      "CPU(X86_64) || CPU(X86)": false
```
Why not do something like?
```suggestion
      default: WebKit::defaultMediaSourcePrefersDecompressionSession()
WebCore:
      default: WebKit::defaultMediaSourcePrefersDecompressionSession()
```

where

```
bool defaultMediaSourcePrefersDecompressionSession()
{
#if CPU(X86_64) || CPU(X86)
    return false;
#else
    return true;
#endif
}
```

aprotyas replied:
(as a workaround, because I do think your initial attempt makes sense)

jyavenard replied:
> (as a workaround, because I do think your initial attempt makes sense)

it's surprising that it gives some static assert error, because a few line below we have exactly the same default being true, and false on other. yet those cause no issue and with a "stable" one.

aprotyas commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -5043,7 +5043,7 @@ MediaSourcePrefersDecompressionSession:
-      default: true
+      default: WebKit::defaultMediaSourcePrefersDecompressionSession()
     WebCore:
       default: true
```
`defaultMediaSourcePrefersDecompressionSession` here too, probably?

jyavenard replied:
No WebKit:: in WebCore bits.

jyavenard committed:
[Power Testing] WebKit.GPU cpu increase in fullscreen video playback on intel[2% power, 12 mins]
https://bugs.webkit.org/show_bug.cgi?id=297102
rdar://157144345

Reviewed by Abrar Rahman Protyasha.

Disable the use a VTDecompressionSession for MSE on Intel platform as it
causes increased power usage.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:

Canonical link: https://commits.webkit.org/298484@main
